### ISSUE TITLE
4.2.0/4.2.1/4.2.2.edge breaks InAppBrowser Window on Android with Platform WebView

### ISSUE THREAD

**Issue by stephan-klein** [AppGyver/steroids#1094]
A similar issue to https://github.com/AppGyver/steroids/issues/1082 occurs when using the Platform WebView:

Its simply reproduceable by calling open(url, "_blank", ...) on InAppBrowser Plugin.
The application will crash on the native side with the following Exception:

> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime: java.lang.ClassCastException: android.webkit.WebView cannot be cast to org.apache.cordova.CordovaWebViewInterface
> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime:     at org.apache.cordova.inappbrowser.InAppBrowser.openUrl(InAppBrowser.java:486)
> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime:     at org.apache.cordova.inappbrowser.InAppBrowser.access$1100(InAppBrowser.java:76)
> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime:     at org.apache.cordova.inappbrowser.InAppBrowser$6.run(InAppBrowser.java:702)
> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime:     at android.app.Activity.runOnUiThread(Activity.java:5524)
> 01-08 19:37:12.628 11349-11349/com.opaygo.guest.scanner E/AndroidRuntime:     at org.apache.cordova.inappbrowser.InAppBrowser.showWebPage(InAppBrowser.java:740)

**Comment by andybbtt11** [comment]
+1

**Comment by harsaharsa** [comment]
Thanks for reporting, we'll look into this!

**Comment by davistan** [comment]
You can try using "_system" instead of "_blank", see whether it helps.

**Comment by andybbtt11** [comment]
This is still a major issue, any updates @harsaharsa ? Thanks

**Comment by stephan-klein** [comment]
Any Updates/Timeline here ?

**Comment by kaiserlos** [comment]
+1

**Comment by stephan-klein** [comment]
As there is no progress on this, i decided to try a workaround / hack

Therefore I forked the original InAppBrowser Plugin from Apache Cordova. To prevent conflicts with the appgyver InAppBrowser I had to rename a lot of stuff. However I could manage to add this plugin to the Appgyver Android Cloud Build, and could use it successfully.

Opening a Browser Window with _blank works now even with Platform Web View.

This is really just a hack I did not test it extensively. Also I might have broken the other Platforms  Versions so only build it for android. Not sure why appgyver initially developed an own fork of InAppBrowser, probably my fork has problems when using inside multiple views in parallel.

Also I had to comment out some code related to Basic Authentication, so sites with Basic Authentication might also fail.

It is available in the Controllers at `cordova.InAppBrowserOrig` and also the `window.open` method will execute the new plugin.

Find it at https://github.com/stephan-klein/cordova-plugin-inappbrowser
